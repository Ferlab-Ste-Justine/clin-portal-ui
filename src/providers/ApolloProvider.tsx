import { ReactElement } from 'react';
import {
  ApolloClient,
  ApolloProvider,
  createHttpLink,
  InMemoryCache,
  NormalizedCacheObject,
} from '@apollo/client';
import { setContext } from '@apollo/client/link/context';
import { GraphqlBackend, GraphqlProvider } from 'providers';
import { appendBearerIfToken } from 'utils/helper';
import { useRpt } from 'hooks/useRpt';
import EnvironmentVariables from 'utils/EnvVariables';

export const ARRANGER_API = EnvironmentVariables.configFor('ARRANGER_API');
export const PROJECT_ID = EnvironmentVariables.configFor('ARRANGER_PROJECT_ID');
export const FHIR_API = EnvironmentVariables.configFor('FHIR_API');

export const ARRANGER_API_PROJECT_URL = `${ARRANGER_API}/${PROJECT_ID}/graphql`;

const fhirLink = createHttpLink({
  uri: `${FHIR_API}/$graphql`,
});

const arrangerLink = createHttpLink({
  uri: ARRANGER_API_PROJECT_URL,
});

const getAuthLink = (token: string) =>
  setContext((_, { headers }) => ({
    headers: {
      ...headers,
      authorization: appendBearerIfToken(token),
    },
  }));

const backendUrl = (backend: GraphqlBackend) =>
  backend === GraphqlBackend.FHIR ? fhirLink : arrangerLink;

const Provider = ({ children, backend = GraphqlBackend.FHIR }: GraphqlProvider): ReactElement => {
  const { loading, rpt } = useRpt();
  if (loading) {
    return <></>;
  }
  const header = getAuthLink(rpt);

  const client: ApolloClient<NormalizedCacheObject> = new ApolloClient({
    cache: new InMemoryCache({ addTypename: false }),
    link: header.concat(backendUrl(backend)),
  });
  return <ApolloProvider client={client}>{children}</ApolloProvider>;
};

export default Provider;
