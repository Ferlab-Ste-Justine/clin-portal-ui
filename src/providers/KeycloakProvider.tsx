import React, { ReactElement } from 'react';
import { AuthClientError, AuthClientEvent } from '@react-keycloak/core/lib/types';
import { ReactKeycloakProvider as KeycloakProvider } from '@react-keycloak/web';
import keycloak from 'auth/keycloak';

import { trackAuthSuccess } from 'services/analytics';
import EnvVariables from 'utils/EnvVariables';

export interface IProvider {
  children: React.ReactNode;
}

const eventLogger = (eventType: AuthClientEvent, error?: AuthClientError) => {
  if (EnvVariables.configFor('ENV') === 'development' && error) {
    console.error('eventLogger ', 'eventType ', eventType);
    console.error('eventLogger ', error);
  }

  if (eventType === 'onAuthSuccess') {
    trackAuthSuccess();
  }
};

const Keycloak = ({ children }: IProvider): ReactElement => (
  <KeycloakProvider authClient={keycloak} onEvent={eventLogger}>
    {children}
  </KeycloakProvider>
);

export default Keycloak;
