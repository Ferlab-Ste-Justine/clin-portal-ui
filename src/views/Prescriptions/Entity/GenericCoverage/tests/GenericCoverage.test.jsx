import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { default as ApolloProvider } from 'providers/ApolloProvider';
import configureStore from 'redux-mock-store';

import GenericCoverage from '../index';

describe('Generic Coverage', () => {
  test('should be as the snapshot', () => {
    const initialState = {};
    const store = configureStore()(initialState);
    const component = renderer.create(
      <Provider store={store}>
        <ApolloProvider>
          <Router>
            <GenericCoverage />
          </Router>
        </ApolloProvider>
      </Provider>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
