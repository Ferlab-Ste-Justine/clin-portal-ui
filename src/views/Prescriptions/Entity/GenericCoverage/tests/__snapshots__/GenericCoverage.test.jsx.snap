// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generic Coverage should be as the snapshot 1`] = `null`;
